builtin/providers/cloudstack/resource_cloudstack_affinity_group.go

```diff
@@ -0,0 +1,136 @@
+package cloudstack
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+func resourceCloudStackAffinityGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudStackAffinityGroupCreate,
+		Read:   resourceCloudStackAffinityGroupRead,
+		Delete: resourceCloudStackAffinityGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceCloudStackAffinityGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	cs := meta.(*cloudstack.CloudStackClient)
+
+	name := d.Get("name").(string)
+	affinityGroupType := d.Get("type").(string)
+
+	log.Printf("[DEBUG] creating affinity group with name %s of type %s", name, affinityGroupType)
+
+	p := cs.AffinityGroup.NewCreateAffinityGroupParams(name, affinityGroupType)
+
+	// Set the description
+	description, hasDescription := d.GetOk("description")
+	if hasDescription {
+		p.SetDescription(description.(string))
+	}
+
+	// If there is a project supplied, we retrieve and set the project id
+	if err := setProjectid(p, cs, d); err != nil {
+		return err
+	}
+
+	r, err := cs.AffinityGroup.CreateAffinityGroup(p)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] New affinity group successfully generated at Cloudstack")
```

Comment:
_nitting_ Maybe this looks cleaner: `log.Printf("[DEBUG] New affinity group successfully created")`